subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/client/GradleProfilerClient.java

```diff
@@ -63,19 +76,44 @@ private void handleGradleProfilerRequest(StudioRequest request, Project project)
 
     private void handleSyncRequest(StudioRequest request, Project project) {
         LOG.info("Received sync request with id: " + request.getId());
+        boolean isStartup = syncCount.getAndIncrement() == 0;
 
         // In some cases sync could happen before we trigger it,
         // for example when we open a project for the first time.
         waitOnPreviousGradleSyncFinish(project);
         waitOnBackgroundProcessesFinish(project);
 
-        Stopwatch stopwatch = Stopwatch.createStarted();
         LOG.info(String.format("[SYNC REQUEST %s] Sync has started%n", request.getId()));
+        Stopwatch stopwatch = isStartup ? startupStopwatch : Stopwatch.createStarted();
+        GradleSyncResult result = isStartup ? getStartupSyncResult(project) : startManualSync(project);
+        LOG.info(String.format("[SYNC REQUEST %s] '%s': '%s'%n", request.getId(), result.getResult(), result.getErrorMessage()));
+        client.send(new StudioSyncRequestCompleted(request.getId(), stopwatch.elapsed(TimeUnit.MILLISECONDS), result.getResult(), result.getErrorMessage()));
+    }
+
+    private GradleSyncResult startManualSync(Project project) {
         GradleSyncResult result = doGradleSync(project);
         waitOnBackgroundProcessesFinish(project);
-        String errorMessage = result.getErrorMessage().isEmpty() ? "no error message" : result.getErrorMessage();
-        LOG.info(String.format("[SYNC REQUEST %s] '%s': '%s'%n", request.getId(), result.getResult(), errorMessage));
-        client.send(new StudioSyncRequestCompleted(request.getId(), stopwatch.elapsed(TimeUnit.MILLISECONDS), result.getResult(), errorMessage));
+        return result;
+    }
+
+    /**
+     * Gets the result of startup sync by checking if there was already any sync done before. Otherwise, it starts a new sync.
+     */
+    private GradleSyncResult getStartupSyncResult(Project project) {
+        ProjectSystemSyncManager.SyncResult lastSyncResult = ProjectSystemUtil.getSyncManager(project).getLastSyncResult();
+        if (lastSyncResult == UNKNOWN) {
+            // Sync was not run before, we need to run it manually
+            return startManualSync(project);
+        }
+        GradleSyncResult result;
+        if (lastSyncResult.isSuccessful() && (lastSyncResult == SKIPPED || lastSyncResult == SKIPPED_OUT_OF_DATE)) {
+            result = new GradleSyncResult(StudioSyncRequestResult.SKIPPED, "");
+        } else if (lastSyncResult.isSuccessful()) {
+            result = new GradleSyncResult(StudioSyncRequestResult.SUCCEEDED, "");
+        } else  {
+            result = new GradleSyncResult(StudioSyncRequestResult.FAILED, "Startup failure");
+        }
+        return result;
```

Comment:
We could simplify this a bit, for example there's no need to check for `isSuccessful()` when dealing with the skipped stated.

```suggestion
        if (lastSyncResult == UNKNOWN) {
            // Sync was not run before, we need to run it manually
            return startManualSync(project);
        } else if (lastSyncResult == SKIPPED || lastSyncResult == SKIPPED_OUT_OF_DATE) {
            return new GradleSyncResult(StudioSyncRequestResult.SKIPPED, "");
        } else if (lastSyncResult.isSuccessful()) {
            return new GradleSyncResult(StudioSyncRequestResult.SUCCEEDED, "");
        } else  {
            return new GradleSyncResult(StudioSyncRequestResult.FAILED, "Startup failure");
        }
```

Reply:
thanks